bftengine/tests/testViewChange/testViewChange.cpp

```diff
@@ -0,0 +1,204 @@
+  ViewNum curView = 0;
+
+  auto* CO = new ClientRequestMsg((uint16_t)1,
+                                  bftEngine::ClientMsgFlag::EMPTY_FLAGS_REQ,
+                                  (uint64_t)1234567,
+                                  kRequestLength,
+                                  (const char*)requestBuffer,
+                                  (uint64_t)1000000);
+  auto* pp = new PrePrepareMsg(0, curView, 1, bftEngine::impl::CommitPath::SLOW, false);
+  pp->addRequest(CO->body(), CO->size());
+  pp->finishAddingRequests();
+
+  PreparePartialMsg* p1 = PreparePartialMsg::create(curView,
+                                                    pp->seqNumber(),
+                                                    replicaConfig[1].replicaId,
+                                                    pp->digestOfRequests(),
+                                                    replicaConfig[1].thresholdSignerForSlowPathCommit);
+  PreparePartialMsg* p2 = PreparePartialMsg::create(curView,
+                                                    pp->seqNumber(),
+                                                    replicaConfig[2].replicaId,
+                                                    pp->digestOfRequests(),
+                                                    replicaConfig[2].thresholdSignerForSlowPathCommit);
+  PreparePartialMsg* p3 = PreparePartialMsg::create(curView,
+                                                    pp->seqNumber(),
+                                                    replicaConfig[3].replicaId,
+                                                    pp->digestOfRequests(),
+                                                    replicaConfig[3].thresholdSignerForSlowPathCommit);
+
+  seqNumInfo_.addSelfMsg(pp);
+  seqNumInfo_.addMsg(p1);
+  seqNumInfo_.addMsg(p2);
+  seqNumInfo_.addMsg(p3);
+
+  PrepareFullMsg pfMsg();
+
+  ViewChangeMsg** viewChangeMsgs = new ViewChangeMsg*[N];
+  for (int i = 0; i < N; i++) {
+    if (i == 1) {
+      viewChangeMsgs[i] = nullptr;
+    } else {
+      viewChangeMsgs[i] = new ViewChangeMsg(i, 1, 0);
+    }
+  }
+
+  // viewChangeMsgs[2]->addElement(*pRepInfo,
+  //                   1,
+  //                   pp->digestOfRequests(),
+  //                   0,
+  //                   true,
+  //                   0,
+  //                   pfMsg.signatureLen(),
+  //                   pfMsg.signatureBody()
+  //                   );
+
+  auto VCS = ViewChangeSafetyLogic(
+      N, F, C, replicaConfig[0].thresholdVerifierForSlowPathCommit, PrePrepareMsg::digestOfNullPrePrepareMsg());
+
+  SeqNum min{}, max{};
+  VCS.computeRestrictions(viewChangeMsgs, 0, min, max, restrictions);
+
+  replica.getInternalThreadPool().stop(true);
+
+  for (int i = 0; i < N; i++) {
+    delete viewChangeMsgs[i];
+    viewChangeMsgs[i] = nullptr;
+  }
+  delete[] viewChangeMsgs;
+}
+
+}  // end namespace
+
+int main(int argc, char** argv) {
```

Comment:
Can you also add one test for `calcLBStableForView()` where it returns a proper sequence number, instead of crashing?